### PR TITLE
Add env vars, Dynamo read access for entitlement

### PR DESCRIPTION
**Pull request by mommothazaz123**
Adds the following TF vars:
- `auth_service_url`: the URL of the auth service endpoint to hit for Discord auth
- `entitlements_user_dynamo_table`: the name of the user entitlements Dynamo table
- `entitlements_entity_dynamo_table`: the name of the entity entitlements Dynamo table
- `entitlements_dynamo_table_prefix`: the common prefix of entitlements Dynamo tables (to grant read access)

And the following secret:
- `avrae/${var.env}/auth-service-secret`: the JWT secret

And grants the following permissions to Avrae and Avrae Nightly:
- `"dynamodb:GetItem", "dynamodb:Query"` on tables with the right prefix

**Before Merge**
- set the TF vars

**Before Run**
- note that running this plan will cause a production bot restart!
- we should merge avrae/avrae/2020w14 to make bot restart lock logic better before run

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -112,6 +112,13 @@ resource "aws_secretsmanager_secret" "avrae_bot_nightly_discord_token" {
   tags        = local.common_tags
 }
 
+# Secrets Manager Secret - Auth Service Secret
+resource "aws_secretsmanager_secret" "auth_service_secret" {
+  name        = "avrae/${var.env}/auth-service-secret"
```
Comment (zmckinnon) [inline main.tf]: I am not sure where this should be or what it should be named.  Right now, it doesn't match what is in the service. 🤔

Comment (Croebh) [reply main.tf]: Clearly it should be 'aws-secret-service-secret'

Comment (zmckinnon) [reply main.tf]: On second thought, we'd just need @Truckboat to connect this to what I have as an environment variable, so that doesn't actually matter.

Comment (mommothazaz123) [reply main.tf]: If the secret's defined in the auth service TF, we can remove this entirely and just grab its ARN, unless there's a cleaner way to do that (deferring to Aaron)

**Pull request merged**